packages/grid/src/HeaderCell.tsx

```diff
@@ -30,25 +38,82 @@ export function HeaderCell<T>(props: HeaderCellProps<T>) {
   const { ref, isFocusableContent, onFocus } =
     useFocusableContent<HTMLTableHeaderCellElement>();
 
+  const { onClickSortColumn, setSortByColumnId, sortOrder, sortByColumnId } =
+    useColumnSortContext();
+
+  const valueAlignRight = align === "right";
+
+  interface HeaderCellSortingIconProps {
+    justify: FlexContentAlignment;
+  }
+
+  const HeaderCellSortingIcon = ({ justify }: HeaderCellSortingIconProps) => {
+    const className = withBaseName("sortingIcon");
+    const icon = (
+      <FlexLayout className={className} justify={justify} aria-hidden>
+        {sortOrder === "asc" && <ArrowUpIcon />}
+        {sortOrder === "desc" && <ArrowDownIcon />}
+      </FlexLayout>
+    );
+
+    return icon;
+  };
+
+  type AriaSortProps = "none" | "ascending" | "descending";
```

Comment:
These should be moved out of the component

Reply:
I assume we could extract this from the standard lib instead of defining it ourselves? Like HTMLTableRowAttributes["aria-sort"] or something like that.

Reply:
I've moved it out of the component instead as was not having much luck on typescript's end using the types `HTMLTableRowElement["ariaSort"]`